src/jetson/main.py

```diff
@@ -4,6 +4,7 @@
 from typing import List, Set, Dict, Tuple, Optional
 
```

Comment:
there is a lot happening in this file. so its very hard to read. A good rule of thumb is to keep class definitions out of the main file. Maybe its time to add that refactor here to make this easier to read.